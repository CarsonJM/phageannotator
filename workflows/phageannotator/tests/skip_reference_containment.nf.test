nextflow_workflow {

    name "Test workflow: PHAGEANNOTATOR"
    script "workflows/phageannotator/main.nf"
    workflow "PHAGEANNOTATOR"
    tag "workflows"
    tag "phageannotator"
    tag "phageannotator_skip_reference"


    test("Parameters: --skip-reference-containment") {

        when {
            workflow {
                """
                input[0] = Channel.of( [ [ id:'test' ], [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz', checkIfExists: true),
                            file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz', checkIfExists:true) ] ] )
                input[1] = Channel.of( [ [ id:'test' ], [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists:true) ] ] )
                """
            }
            params {
                skip_reference_containment  = true
                outdir                      = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
