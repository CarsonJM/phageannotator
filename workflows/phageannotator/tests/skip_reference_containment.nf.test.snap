{
    "Parameters: --skip-reference-containment": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.tsv:md5,78054c131c9cbe3d696b7c29c287ea7b"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.IS_gene_info.tsv:md5,0d05798c82a35fd4fd2787ff1cb9b607"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.IS_gene_info.tsv:md5,0d05798c82a35fd4fd2787ff1cb9b607"
                    ]
                ],
                "11": [
                    "versions.yml:md5,00a12ed45ed179a55e30c05dbb3d31f3",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,1097ac7d62d13e9e74916d9dc1fabb7b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,2d1864a0523e2738f150eaf87d4472d5",
                    "versions.yml:md5,3c8b8ac5a86d6d7fba86fc61f4d5fca9",
                    "versions.yml:md5,3d3432539468e0fae3b05d63cd5237d0",
                    "versions.yml:md5,42ce3c1803e6f3030b60d1d4610b629d",
                    "versions.yml:md5,45fa41ba465519ccbfe26974f9aebcc4",
                    "versions.yml:md5,4698f3f4cdba1748298d6fc664fe55d4",
                    "versions.yml:md5,478c8fdcebf1357f40c26d7991d85eb4",
                    "versions.yml:md5,5074b29ede738ba6d39ef056e1cde929",
                    "versions.yml:md5,57a9c1152c52dc32d2023ce8e2b939ab",
                    "versions.yml:md5,57a9c1152c52dc32d2023ce8e2b939ab",
                    "versions.yml:md5,6bf7a212d54ddc08bb3d6c7ce9f4cddb",
                    "versions.yml:md5,6bf7a212d54ddc08bb3d6c7ce9f4cddb",
                    "versions.yml:md5,6fd21ed738b35b25a1f47b313be8f47e",
                    "versions.yml:md5,755876cdfeb32585833c108ed6635839",
                    "versions.yml:md5,755876cdfeb32585833c108ed6635839",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,76fc7f1f019facce5a0af4d78f01c624",
                    "versions.yml:md5,76fc7f1f019facce5a0af4d78f01c624",
                    "versions.yml:md5,78b447897aac9e6a891f06a6c4af0e99",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,8f4cfce986149a10f9c2029b48c28810",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,995518cecee8eed38a4f357b7f57fd33",
                    "versions.yml:md5,a6aa8efc5dbcafe7129f58ed120eaa93",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,bb255696c1826fcc93261acc5de1cd07",
                    "versions.yml:md5,c7c1550786083ee8f3264b19ccf0fcbb",
                    "versions.yml:md5,c7c1550786083ee8f3264b19ccf0fcbb",
                    "versions.yml:md5,c97cb7740ce1aa3cf22cb6e8be2d62ce",
                    "versions.yml:md5,f383a8d4db50b157e4a6a0ef4670fbc6",
                    "versions.yml:md5,fc95484076ab0130345b4f7ab7f2e37a"
                ],
                "2": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.tsv:md5,ddbe23633823e639a6087e3c420d7ce2"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,98f190f40f0e135847f8adacf04e40d1"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "Host_prediction_to_genus_m90.csv:md5,528dcefb6a6c743b43f0dbc40bec3692"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.bacphlip:md5,ca1ec76d35f9c090200871fa302441f7"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.faa.gz:md5,ba02f7f691834c78746d404d5408b693"
                    ]
                ],
                "alignment_results_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_alignment_results.tsv:md5,98f190f40f0e135847f8adacf04e40d1"
                    ]
                ],
                "anicluster_reps_fna_gz": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "bacphlip_lifestyle_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives.fna.bacphlip:md5,ca1ec76d35f9c090200871fa302441f7"
                    ]
                ],
                "filtered_viruses_fna_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                    ]
                ],
                "host_predictions_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "Host_prediction_to_genus_m90.csv:md5,528dcefb6a6c743b43f0dbc40bec3692"
                    ]
                ],
                "instrain_gene_info": [
                    [
                        {
                            "id": "test2"
                        },
                        "test2.IS_gene_info.tsv:md5,0d05798c82a35fd4fd2787ff1cb9b607"
                    ],
                    [
                        {
                            "id": "test"
                        },
                        "test.IS_gene_info.tsv:md5,0d05798c82a35fd4fd2787ff1cb9b607"
                    ]
                ],
                "marker_taxonomy_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples_anicluster_representatives_virus_summary.tsv:md5,ae6d66b39f6ea43628db461e42012027"
                    ]
                ],
                "prodigalgv_proteins_faa_gz": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.faa.gz:md5,ba02f7f691834c78746d404d5408b693"
                    ]
                ],
                "reference_containment_tsv": [
                    
                ],
                "versions": [
                    "versions.yml:md5,00a12ed45ed179a55e30c05dbb3d31f3",
                    "versions.yml:md5,04bf25e5231ce1f8077581bb7a2c07f4",
                    "versions.yml:md5,1097ac7d62d13e9e74916d9dc1fabb7b",
                    "versions.yml:md5,1439e948f383899aaccbf49506923b04",
                    "versions.yml:md5,2d1864a0523e2738f150eaf87d4472d5",
                    "versions.yml:md5,3c8b8ac5a86d6d7fba86fc61f4d5fca9",
                    "versions.yml:md5,3d3432539468e0fae3b05d63cd5237d0",
                    "versions.yml:md5,42ce3c1803e6f3030b60d1d4610b629d",
                    "versions.yml:md5,45fa41ba465519ccbfe26974f9aebcc4",
                    "versions.yml:md5,4698f3f4cdba1748298d6fc664fe55d4",
                    "versions.yml:md5,478c8fdcebf1357f40c26d7991d85eb4",
                    "versions.yml:md5,5074b29ede738ba6d39ef056e1cde929",
                    "versions.yml:md5,57a9c1152c52dc32d2023ce8e2b939ab",
                    "versions.yml:md5,57a9c1152c52dc32d2023ce8e2b939ab",
                    "versions.yml:md5,6bf7a212d54ddc08bb3d6c7ce9f4cddb",
                    "versions.yml:md5,6bf7a212d54ddc08bb3d6c7ce9f4cddb",
                    "versions.yml:md5,6fd21ed738b35b25a1f47b313be8f47e",
                    "versions.yml:md5,755876cdfeb32585833c108ed6635839",
                    "versions.yml:md5,755876cdfeb32585833c108ed6635839",
                    "versions.yml:md5,769114e3bb009aceb4763a1a0fa80d5a",
                    "versions.yml:md5,76fc7f1f019facce5a0af4d78f01c624",
                    "versions.yml:md5,76fc7f1f019facce5a0af4d78f01c624",
                    "versions.yml:md5,78b447897aac9e6a891f06a6c4af0e99",
                    "versions.yml:md5,7e54b5f822df67b99faf67629e81963f",
                    "versions.yml:md5,8aebba726df29aad131772449019ee78",
                    "versions.yml:md5,8f4cfce986149a10f9c2029b48c28810",
                    "versions.yml:md5,987d2e54a85311b917599dc41c90418c",
                    "versions.yml:md5,995518cecee8eed38a4f357b7f57fd33",
                    "versions.yml:md5,a6aa8efc5dbcafe7129f58ed120eaa93",
                    "versions.yml:md5,a856eefde99260259440d942c1d56ff4",
                    "versions.yml:md5,bb255696c1826fcc93261acc5de1cd07",
                    "versions.yml:md5,c7c1550786083ee8f3264b19ccf0fcbb",
                    "versions.yml:md5,c7c1550786083ee8f3264b19ccf0fcbb",
                    "versions.yml:md5,c97cb7740ce1aa3cf22cb6e8be2d62ce",
                    "versions.yml:md5,f383a8d4db50b157e4a6a0ef4670fbc6",
                    "versions.yml:md5,fc95484076ab0130345b4f7ab7f2e37a"
                ],
                "virus_classification_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.tsv:md5,78054c131c9cbe3d696b7c29c287ea7b"
                    ]
                ],
                "virus_quality_tsv": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.tsv:md5,ddbe23633823e639a6087e3c420d7ce2"
                    ]
                ]
            }
        ],
        "timestamp": "2023-12-07T10:45:22.619730467"
    }
}